packages/extensionmanager/src/companions.tsx

```diff
@@ -96,6 +96,27 @@ export interface IJupyterLabPackageData {
   };
 }
 
+function getInstallCommands(info: IInstallInfo) {
+  let commands = Array<string>();
+  if (info.overrides?.pip?.name) {
+    commands.push(`pip install ${info.overrides.pip.name}`);
+  } else {
+    const pip = info.managers.find(s => s === 'pip');
+    if (pip) {
+      commands.push(`pip install ${info.base.name}`);
+    }
+  }
+  if (info.overrides?.conda?.name) {
+    commands.push(`conda install -c conda-forge ${info.overrides.conda.name}`);
+  } else {
+    const conda = info.managers.find(s => s === 'conda');
+    if (conda) {
+      commands.push(`conda install -c conda-forge ${info.base.name}`);
+    }
+  }
+  return commands;
+}
+
```

Comment:
I had an idea here on a slight refactor. It differs slightly in semantics:

1. Only lookup commands for managers in `info.managers` instead of `info.overrides`. This means if you had `pip` in `overrides` but not in `managers` it wouldn't be printed. I *think* this is more in line with the documented behavior. But also maybe it's simply a "undefined behavior" if we have an overrides but is not listed in a manager?
2. Logs a warning if there is a manager listed that isn't `pip` or `conda`
3. Logs a warning if there is no default name listed and no name for the installer.

Also I figured out how you do optional chaining after an index type... You have to do `name?.[index]`, notice the dot.

```suggestion
// Mapping of manager name to function that take name and gives command
const managerCommand: { [key: string]: (name: string) => string } = {
  pip: name => `pip install ${name}`,
  conda: name => `conda install -c conda-forge ${name}`
};

function getInstallCommands(info: IInstallInfo) {
  let commands = Array<string>();
  for (const manager of info.managers) {
    const name = info.overrides?.[manager]?.name ?? info.base.name;
    if (!name) {
      console.warn(`No package name found for manager ${manager}`);
      continue;
    }
    const command = managerCommand[manager]?.(name);
    if (!command) {
      console.warn(`Don't know how to install packages for manager ${manager}`);
    }
    commands.push(command);
  }
  return commands;
}

```

Reply:
Yep, I think your logic makes more sense. `overrides` is supposed to override something.

Have you tested this change locally?

Good idea to print a warning in the console.

Reply:
Yeah, I wrote this locally and it seems to work the same. (I didn't test the warning paths though).

Reply:
Cool. I will merge your changes.